ui/src/containers/UserProfile/index.tsx

```diff
@@ -15,16 +15,48 @@ import './UserProfile.css';
 const UserProfile: React.FC = () => {
   const { user } = useMst();
 
+  const [refreshId, setRefreshId] = useState<number>();
+  const [accessId, setAccessId] = useState<number>();
+
+  useEffect(() => {
+    const triggerInterval = () => {
+      const accessTokenInterval = user.accessTokenInfo.expiresAt * 1000 - new Date().getTime();
+      const refreshTokenInterval = user.refreshTokenInfo.expiresAt * 1000 - new Date().getTime();
+
+      // setInterval for refresh the refresh token before 20 seconds of expiry time
+      const tempRefreshId = window.setInterval(() => {
+        user.updateRefreshToken();
+      }, refreshTokenInterval - 2000);
+      setRefreshId(tempRefreshId);
+
+      // setInterval for refresh the access token before 20 seconds of expiry time
+      const tempAccessId = window.setInterval(() => {
+        user.updateAccessToken();
+      }, accessTokenInterval - 2000);
+      setAccessId(tempAccessId);
+    };
+
+    triggerInterval();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [isOpen, set] = useState(false);
 
+  const hubLogout = () => {
```

Comment:
Should we move this function to store :thinking: ?

Reply:
`localStore.clear()` was not working in the store so I kept here

Reply:
Other than `localstorage` can be just kept other can me moved right ???

Reply:
yeah that's already in the store